controllers/storageclient_controller.go

```diff
@@ -210,9 +217,25 @@ func (r *StorageClientReconciler) reconcilePhases() (ctrl.Result, error) {
 			if err := r.reconcileClusterResourceQuota(clusterResourceQuotaSpec); err != nil {
 				return reconcile.Result{}, err
 			}
+		case "Secret":
+			if eResource.Name == fmt.Sprintf("noobaa-remote-join-secret-%s", r.storageClient.Status.ConsumerID) {
```

Comment:
Why match the exact name, use `strings.hasPrefix` instead.
But also, is there a reason we need the suffix at all? there is only going to be a single join secret on this cluster

Reply:
Wil change the prefix matching in updated commit
This is to keep the code viable for multiple noobaa clients in one application cluster
Ref: https://github.com/red-hat-storage/ocs-client-operator/issues/189